Backpack/src/main/java/net/skyscanner/backpack/button/BpkButton.kt

```diff
@@ -25,8 +26,26 @@ open class BpkButton @JvmOverloads constructor(
   defStyleAttr: Int = 0
 ) : AppCompatButton(context, attrs, defStyleAttr) {
 
+  @IntDef(START, END, ICON_ONLY)
+
+  annotation class IconPosition
+
+  companion object {
+    const val START = 0
+    const val END = 1
+    const val ICON_ONLY = 2
```

Comment:
Why do we need this? Can't we just ignore `START` or `END` if a text isn't set? That's what we do on iOS

Reply:
There were a few problems with that:
1. Its possible to set both start and end at the same time with no priority to either.
2. In case of icon only, the padding values are different to make it look round. It was depending on checking the text length. With 'iconOnly` any extra text is force removed

Reply:
How can you set start and end at the same time?

Reply:
It was possible by setting both `iconStart` and `iconEnd` properties for the same button. With this PR it would not be allowed anymore. This also aligns with the button api on RN